tests/php/ORM/DBReplicaTest.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace SilverStripe\ORM\Tests;
+
+use ReflectionClass;
+use SilverStripe\Dev\FunctionalTest;
+use SilverStripe\ORM\DB;
+use SilverStripe\Control\Director;
+use SilverStripe\Security\Security;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\ORM\Tests\DBReplicaTest\TestController;
+use SilverStripe\ORM\Tests\DBReplicaTest\TestObject;
+use SilverStripe\Security\Group;
+use SilverStripe\Security\Member;
+use SilverStripe\ORM\DataQuery;
+use SilverStripe\Security\Permission;
+
+/**
+ * Test for DB replica functionality
+ *
+ * Note that a lot of these methods test mutiple things at once. This is to increase the run speed of the tests
+ * because they're all quite slow due to the need to create and tear-down fixtures for every test
+ */
+class DBReplicaTest extends FunctionalTest
+{
+    protected static $extra_dataobjects = [
+        TestObject::class,
+    ];
+
+    protected static $fixture_file = 'DBReplicaTest.yml';
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->setupConfigsAndConnections(true);
+        // Allow using replicas - this is disabled by default in SapphireTest::setUpBeforeClass()
+        DB::setCanUseReplicas(true);
+        // Reset mustUsePrimary after using mutable sql to create yml fixtures
+        // and also because by default an ADMIN user is logged in when using fixtures in SapphireTest::setUp()
+        // and also prevent tests from affecting subsequent tests
+        (new ReflectionClass(DB::class))->setStaticPropertyValue('mustUsePrimary', false);
+    }
+
+    protected function tearDown(): void
+    {
+        $this->setupConfigsAndConnections(false);
+        DB::setCanUseReplicas(false);
+        parent::tearDown();
+    }
+
+    /**
+     * Using reflection, set DB::configs and DB::connections with a fake a replica connection
+     * that points to the same connection as the primary connection.
+     */
+    private function setupConfigsAndConnections($includeReplica = true)
+    {
+        $reflector = new ReflectionClass(DB::class);
+        $defaultConfig = DB::getConfig('default');
+        $configs = ['default' => $defaultConfig];
+        if ($includeReplica) {
+            $configs['primary'] = $defaultConfig;
+            $configs['replica_01'] = $defaultConfig;
+        }
+        $reflector->setStaticPropertyValue('configs', $configs);
+        // Make a call to the database to ensure the connection is established
+        TestObject::get()->count();
```

Comment:
Why not call `DB::get_conn()` or `DB::connect()` which will both ensure the connection is established?
And for that matter... why do we need to ensure the connection is established at this stage?

Reply:
Removed